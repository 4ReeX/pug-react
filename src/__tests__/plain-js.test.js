import testHelper from './test-helper';

describe('code', () => {
  testHelper(__dirname + '/plain-js-code.input.js');
});

describe('code-multiline', () => {
  testHelper(__dirname + '/plain-js-code-multiline.input.js');
});

describe('if', () => {
  testHelper(__dirname + '/plain-js-if.input.js');
});

describe('each', () => {
  testHelper(__dirname + '/plain-js-each.input.js');
});

describe('each-multiple', () => {
  testHelper(__dirname + '/plain-js-each-multiple.input.js');
});

describe('while', () => {
  testHelper(__dirname + '/plain-js-while.input.js');
});

describe('case', () => {
  testHelper(__dirname + '/plain-js-case.input.js');
});
