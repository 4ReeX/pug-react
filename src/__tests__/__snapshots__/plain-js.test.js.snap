// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`case JavaScript output: transformed source code 1`] = `
"const n = 0;

module.exports =
  n === 1 ? (
    <p key=\\"pug:0:0\\">One</p>
  ) : n === 2 ? (
    <p key=\\"pug:1:0\\">Two</p>
  ) : null;
"
`;

exports[`case html output: generated html 1`] = `null`;

exports[`case static html output: static html 1`] = `""`;

exports[`code JavaScript output: transformed source code 1`] = `
"const log = text => text;

module.exports = (log(\\"Hello\\"), null);
"
`;

exports[`code html output: generated html 1`] = `null`;

exports[`code static html output: static html 1`] = `""`;

exports[`code-multiline JavaScript output: transformed source code 1`] = `
"let _greeting;

const log = text => text;

module.exports = (
  <React.Fragment>
    {((_greeting = \\"hello\\"), null)}
    {(log(_greeting), null)}
  </React.Fragment>
);
"
`;

exports[`code-multiline html output: generated html 1`] = `null`;

exports[`code-multiline static html output: static html 1`] = `""`;

exports[`each JavaScript output: transformed source code 1`] = `
"module.exports = [1, 2, 3].map(item => {
  return <div key={item}>{item}</div>;
});
"
`;

exports[`each html output: generated html 1`] = `
Array [
  <div>
    1
  </div>,
  <div>
    2
  </div>,
  <div>
    3
  </div>,
]
`;

exports[`each static html output: static html 1`] = `"<div>1</div><div>2</div><div>3</div>"`;

exports[`each-multiple JavaScript output: transformed source code 1`] = `
"module.exports = (
  <React.Fragment>
    {[1, 2, 3].map(item => {
      return <div key={item}>{item}</div>;
    })}
    {[1, 2, 3].map(item => {
      return <div key={item}>{item}</div>;
    })}
  </React.Fragment>
);
"
`;

exports[`each-multiple html output: generated html 1`] = `
Array [
  <div>
    1
  </div>,
  <div>
    2
  </div>,
  <div>
    3
  </div>,
  <div>
    1
  </div>,
  <div>
    2
  </div>,
  <div>
    3
  </div>,
]
`;

exports[`each-multiple static html output: static html 1`] = `"<div>1</div><div>2</div><div>3</div><div>1</div><div>2</div><div>3</div>"`;

exports[`if JavaScript output: transformed source code 1`] = `
"module.exports = false ? <p key=\\"pug:0:0\\">Truthy</p> : null;
"
`;

exports[`if html output: generated html 1`] = `null`;

exports[`if static html output: static html 1`] = `""`;

exports[`while JavaScript output: transformed source code 1`] = `
"let n = 0;

module.exports = (_pug_nodes => {
  while (n < 3) {
    _pug_nodes[_pug_nodes.length] = <div key={n}>{n++}</div>;
  }

  return _pug_nodes;
})([]);
"
`;

exports[`while html output: generated html 1`] = `
Array [
  <div>
    0
  </div>,
  <div>
    1
  </div>,
  <div>
    2
  </div>,
]
`;

exports[`while static html output: static html 1`] = `"<div>0</div><div>1</div><div>2</div>"`;
